@here/harp-mapview/lib/text/TextElementsRenderer.ts

```diff
@@ -1646,17 +1635,41 @@ export class TextElementsRenderer {
             const textIsOptional: boolean =
                 pointLabel.poiInfo !== undefined && pointLabel.poiInfo.textIsOptional === true;
 
-            const textIsFadingIn = textRenderState!.isFadingIn();
-            const textIsFadingOut = textRenderState!.isFadingOut();
-            const textSpaceAvailable = !this.m_screenCollisions.isAllocated(tempBox2D);
-            const textVisible =
-                groupState.visited &&
-                (pointLabel.textMayOverlap ||
-                    textSpaceAvailable ||
-                    textIsFadingIn ||
-                    textIsFadingOut);
-
-            if (textVisible) {
+            const textSpaceAvailable =
+                pointLabel.textMayOverlap || !this.m_screenCollisions.isAllocated(tempBox2D);
+            const textRejected = !groupState.visited || !textSpaceAvailable;
+
+            if (textRejected && !labelState.visible) {
+                if (placementStats) {
+                    placementStats.numPoiTextsInvisible++;
+                }
+                return false;
+            }
+
+            const textWillFadeOut =
+                textRejected && textRenderState!.isVisible() && !textRenderState!.isFadingOut();
+
+            if (textWillFadeOut) {
```

Comment:
as mentioned, I don't see the need to separate this from its definition.

Reply:
what do you mean? getting rid of the constant and putting the expression directly in the if like this? :

`if (textRejected && textRenderState!.isVisible() && !textRenderState!.isFadingOut()) {...`

I find that giving a name to the expression makes it more understandable.